### ISSUE TITLE
CUDNN_STATUS_BAD_PARAM with LSTM/RNN in 1.6.0 with autocast

### ISSUE THREAD

**Issue by DietDietDiet** [pytorch/pytorch#43322]
## 🐛 Bug

I met the following error when I was using pytorch 1.6 with LSTM
RuntimeError: cuDNN error: CUDNN_STATUS_BAD_PARAM

## To Reproduce

I think the root cause was that I was using the mixed precision training mode in pytorch 1.6 following https://pytorch.org/docs/stable/notes/amp_examples.html
and the datatype might be converted to fp16 which causes this error.
When I train my model with full precision, it worked normally. 

cc @csarofeen @ptrblck @zou3519

**Comment by ptrblck** [comment]
Thanks for reporting this issue.
Could you post a code snippet to reproduce this issue as well as your system setup?

You can use the script, which is shown when creating a new issue:
>
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

CC @mcarilli

**Comment by DietDietDiet** [comment]
the training code is like the following,
        with autocast():
            output = model(input)
            loss = loss_fn(output, target)

        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()
and the model definition is

**Comment by DietDietDiet** [comment]
below is the training part,        
 with autocast():
            output = model(input)
            loss = loss_fn(output, target)
        scaler.scale(loss).backward()
        scaler.step(optimizer)
        scaler.update()

**Comment by DietDietDiet** [comment]
below is the training part,
with autocast():
    output = model(input)
    loss = loss_fn(output, target)
scaler.scale(loss).backward()
scaler.step(optimizer)
scaler.update()

the model definition is that
class BidirectionalLSTM(nn.Module):

    def __init__(self, input_size, hidden_size, output_size):
        super(BidirectionalLSTM, self).__init__()
        self.rnn = nn.LSTM(input_size, hidden_size, bidirectional=True, batch_first=True)
        self.linear = nn.Linear(hidden_size * 2, output_size)

    def forward(self, input):
        """
        input : visual feature [batch_size x T x input_size]
        output : contextual feature [batch_size x T x output_size]
        """
        self.rnn.flatten_parameters()
        recurrent, _ = self.rnn(input)  # batch_size x T x input_size -> batch_size x T x (2*hidden_size)
        output = self.linear(recurrent)  # batch_size x T x output_size
        return output

**Comment by ptrblck** [comment]
Thanks for the code snippet.
Unfortunately, I cannot reproduce it using some random inputs and random initialization of the model:
```python
model = BidirectionalLSTM(10, 10, 10).cuda()
optimizer = torch.optim.Adam(model.parameters(), lr=1e-3)
input = torch.randn(16, 10, 10, device='cuda')
loss_fn = nn.CrossEntropyLoss()
target = torch.randint(0, 10, (16, 10), device='cuda')
scaler = torch.cuda.amp.GradScaler()

with torch.cuda.amp.autocast():
    output = model(input)
    loss = loss_fn(output, target)
    scaler.scale(loss).backward()
    scaler.step(optimizer)
    scaler.update()
```
Could you post the training setup, which is causing this issue, as well as the your environment (CUDA, cudnn, PyTorch versions etc.)?

**Comment by mcarilli** [comment]
Probable duplicate of https://github.com/pytorch/pytorch/issues/36428, fixed in master by https://github.com/pytorch/pytorch/pull/42385.  Unfortunately the fix didn't make 1.6.

Closing for now.  If you can try an updated master build or nightly pip/conda package and still see the error, please reopen.
